packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/TerariumAsset.java

```diff
@@ -77,44 +80,53 @@ protected TerariumAsset cloneSuperFields(final TerariumAsset asset) {
 		return asset;
 	}
 
-	public String serializeWithoutTerariumFields() {
-		final ObjectMapper mapper = new ObjectMapper();
-		mapper.setConfig(mapper.getSerializationConfig().with(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY));
-		final ObjectNode objectNode = mapper.convertValue(this, ObjectNode.class);
-		objectNode.remove("id");
-		objectNode.remove("createdOn");
-		objectNode.remove("updatedOn");
-		objectNode.remove("deletedOn");
-		objectNode.remove("name");
-		objectNode.remove("description");
-		objectNode.remove("temporary");
-		objectNode.remove("publicAsset");
-		objectNode.remove("fileNames");
-		objectNode.remove("userId");
-
-		// Remove the field metadata.description as well
-		final JsonNode metadata = objectNode.get("metadata");
-		if (metadata != null) {
-			((ObjectNode) metadata).remove("description");
+	public static void removeFieldsWithKeys(ObjectNode objectNode, List<String> keys) {
+		Iterator<String> keysIterator = objectNode.fieldNames();
+		while (keysIterator.hasNext()) {
+			String key = keysIterator.next();
+			if (keys.contains(key)) {
+				keysIterator.remove();
+			} else {
+				JsonNode node = objectNode.get(key);
+				if (node.isObject()) {
+					removeFieldsWithKeys((ObjectNode) node, keys);
+				}
+			}
 		}
-		objectNode.set("metadata", metadata);
-
-		return objectNode.toString();
 	}
 
-	public String serializeWithoutTerariumFieldsKeepId() {
+	public String serializeWithoutTerariumFields(
```

Comment:
Could you add some JavaDoc so it's easier to read the refactored code done in the controllers?

Reply:
added